internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,274 @@
+
+			id := filesystems.NewFileSystemID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := filesystems.FileSystemResource{
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				Properties: filesystems.FileSystemResourceProperties{
+					AdminPassword:     config.AdminPassword,
+					AvailabilityZone:  pointer.To(config.AvailabilityZone),
+					DelegatedSubnetId: config.DelegatedSubnetId,
+					InitialCapacity:   config.InitialCapacity,
+					StorageSku:        filesystems.StorageSku(config.StorageSku),
+					UserDetails: filesystems.UserDetails{
+						Email: config.UserEmailAddress,
+					},
+					MarketplaceDetails: filesystems.MarketplaceDetails{
+						OfferId:     offerId,
+						PlanId:      config.MarketplacePlanId,
+						PublisherId: publisherId,
+					},
+				},
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, payload); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r FileSystemResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+			schema := FileSystemResourceSchema{}
+
+			id, err := filesystems.ParseFileSystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config FileSystemResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+			schema.AdminPassword = config.AdminPassword
+			schema.UserEmailAddress = config.UserEmailAddress
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+			if model := resp.Model; model != nil {
+				schema.Name = id.FileSystemName
+				schema.ResourceGroupName = id.ResourceGroupName
+				schema.AvailabilityZone = pointer.From(model.Properties.AvailabilityZone)
+				schema.InitialCapacity = model.Properties.InitialCapacity
+				schema.MarketplacePlanId = model.Properties.MarketplaceDetails.PlanId
+				schema.StorageSku = string(model.Properties.StorageSku)
+				schema.Location = location.Normalize(model.Location)
+				schema.Tags = tags.Flatten(model.Tags)
+
+				subnetId, err := commonids.ParseSubnetID(model.Properties.DelegatedSubnetId)
+				if err != nil {
+					return err
+				}
+				schema.DelegatedSubnetId = subnetId.ID()
+			}
```

Comment:
this'd want to become:

```suggestion
			schema.Name = id.FileSystemName
			schema.ResourceGroupName = id.ResourceGroupName
			
			if model := resp.Model; model != nil {
				if props := model.Properties; props != nil {
					schema.AvailabilityZone = pointer.From(props.AvailabilityZone)
					schema.InitialCapacity = props.InitialCapacity
					if props.MarketplaceDetails != nil {
						schema.MarketplacePlanId = props.MarketplaceDetails.PlanId
					}
					schema.StorageSku = string(props.StorageSku)
					schema.Location = location.Normalize(model.Location)
					schema.Tags = tags.Flatten(model.Tags)
	
					subnetId, err := commonids.ParseSubnetIDInsensitively(props.DelegatedSubnetId)
					if err != nil {
						return err
					}
					schema.DelegatedSubnetId = subnetId.ID()
				}
```

1. We should nil-check `model.Properties`
2. We should parse the IDs from the API response insensitively in the Read function
3. We can avoid a few crashes here
4. Fields from the Resource ID can be set outside of the `model` check

Reply:
the `model.Properties` and `model.Properties.MarketplaceDetails` are not pointers, so we do not need to check nil.